### PR TITLE
[fix] fix shift error when fuzzing accumulator schema

### PR DESCRIPTION
**Pull request by lightmark**
fix https://github.com/libra/libra/pull/5800

### CODE REVIEW

File: types/src/proof/position/mod.rs [hunk 1]
```diff
@@ -61,10 +61,10 @@ impl Position {
 
     /// pos count start from 0 on each level
     pub fn from_level_and_pos(level: u32, pos: u64) -> Self {
-        precondition!(level < 63);
+        precondition!(level < 64);
         assume!(1u64 << level > 0); // bitwise and integer operations don't mix.
         let level_one_bits = (1u64 << level) - 1;
-        let shifted_pos = pos << (level + 1);
+        let shifted_pos = pos.wrapping_shl(level + 1);
```
Comment (msmouse) [inline types/src/proof/position/mod.rs:67]: Could you elaborate the reason for this change? I don't see it making any difference for any valid input; if it's invalid input, I think panicking is better than swallowing the error?

Comment (lightmark) [reply types/src/proof/position/mod.rs:67]: if level == 63, the normal << would throw overflow error. but level 63 is valid.

Comment (msmouse) [reply types/src/proof/position/mod.rs:67]: K. It'll make more sense if we panic or error out on a pos too large for the level, but your argument holds.

**Pull request merged**